### ISSUE TITLE
Versions over 0.3.8 Show Unavailable - Furnace

### ISSUE THREAD

**Issue by flangflurply** [dlarrick/hass-kumo#136]
I have a Mitsubishi heat pump furance (not a mini-split system) with the PAC-USWHS002-WF-2 wifi controller. Version 0.3.8 works great and properly displays the information and is controllable via home assistant.

If I upgrade to any version above 0.3.8, I completely lose the ability to connect to the heat pump on any level, forcing me to downgrade to regain access. This has occurred in 0.3.9, 0.3.10 and now 0.3.11.

Per the logs (I'd copy/paste log info but I just got it working again after an upgrade attempt), the integration can't pull information from the unit, despite it being controllable via the app while I was getting the error. It shows, at all times, as Unavailable.

Things I tried while upgraded to the newer versions:
- Reload integration
- Add IP manually and reload integration
- Restart HA
- Reboot HA
- Delete integration and reload (this caused the unit to attempt to connect to a local network IP address that the unit hasn't been on for over a year now - on a completely different subnet than the heat pump is currently - which is baffling). Continued not working after manually deleting the cache file with the outdated IP address and manually entering it in and reloading the integration again.
- Reboot all networking appliances

Are the furnace models that aren't mini splits just no longer supported after 0.3.8? Let me know what information you need me to provide as I'd like to update.

**Comment by dlarrick** [comment]
Nothing was intentionally changed to disable furnace models. But there's nothing I can do directly to diagnose this without having access to a furnace model.

The only change from 0.3.8 to 0.3.9 is an upgrade to pykumo version: https://github.com/dlarrick/hass-kumo/compare/v0.3.8...v0.3.9

And the only change in the relevant pykumo versions was a fix for error handling when certain fields are not present in the response from the Kumo Cloud service: https://github.com/dlarrick/pykumo/compare/v0.3.7...v0.3.8

I would suggest going through the steps under https://github.com/dlarrick/pykumo?tab=readme-ov-file#interactive-use in an interactive Python session. Use a virtualenv to be able to install the older pykumo without interfering with your HomeAssistant installation. See if you can spot the difference.

The other thing I might recommend is to reinstall the older version of hass-kumo with the "prefer cache" option turned on. Then edit the resulting kumo_cache.json file to correct the IP address. Then upgrade to latest hass-kumo and restart.

Also look at https://github.com/dlarrick/pykumo?tab=readme-ov-file#interactive-use to try to use the Installer Settings to get Kumo Cloud to re-synchronize your unit's IP address.

**Comment by flangflurply** [comment]
> The other thing I might recommend is to reinstall the older version of hass-kumo with the "prefer cache" option turned on. Then edit the resulting kumo_cache.json file to correct the IP address. Then upgrade to latest hass-kumo and restart.

So this one I did try, and it did not work. It stopped giving me the IP address error, so that resolved, but then just gave an error that it couldn't read any data from the furnace (I want to say the specific error was not being able to read temp data).

I'll try the forced IP resync, but as I don't believe the error is actually the IP address (given that it stopped giving me the IP error and just stopped working entirely when I gave the correct IP), I don't believe that will fix it. But I'll still try as I'd like to get that IP address banished from existence so thank you for the link either way!

Either way, I'll try the upgrade again this weekend and will collect proper logs and attach, but it's been consistently every version since 0.3.9, and downgrading to 0.3.8 works each time.
